crates/js-sys/src/lib.rs

```diff
@@ -463,6 +503,152 @@ extern "C" {
+        /// until the modified value is written back.
+        ///
+        /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics/and)
+        #[wasm_bindgen(static_method_of = Atomics, catch)]
+        pub fn and(typed_array: &JsValue, index: u32, value: f64) -> Result<f64, JsValue>;
+
+        /// The static `Atomics.compareExchange()` method exchanges a given
+        /// replacement value at a given position in the array, if a given expected
+        /// value equals the old value. It returns the old value at that position
+        /// whether it was equal to the expected value or not.
+        /// This atomic operation guarantees that no other write happens
+        /// until the modified value is written back.
+        ///
+        /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics/compareExchange)
+        #[wasm_bindgen(static_method_of = Atomics, catch, js_name = compareExchange)]
+        pub fn compare_exchange(
+            typed_array: &JsValue,
+            index: u32,
+            expected_value: f64,
+            replacement_value: f64,
+        ) -> Result<f64, JsValue>;
+
+        /// The static `Atomics.exchange()` method stores a given value at a given
+        /// position in the array and returns the old value at that position.
+        /// This atomic operation guarantees that no other write happens
+        /// until the modified value is written back.
+        ///
+        /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics/exchange)
+        #[wasm_bindgen(static_method_of = Atomics, catch)]
+        pub fn exchange(typed_array: &JsValue, index: u32, value: f64) -> Result<f64, JsValue>;
+
+        /// The static `Atomics.isLockFree()` method is used to determine
+        /// whether to use locks or atomic operations. It returns true,
+        /// if the given size is one of the `BYTES_PER_ELEMENT` property
+        /// of integer `TypedArray` types.
+        ///
+        /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics/isLockFree)
+        #[wasm_bindgen(static_method_of = Atomics, js_name = isLockFree)]
+        pub fn is_lock_free(size: u32) -> bool;
+
+        /// The static `Atomics.load()` method returns a value at a given
+        /// position in the array.
+        ///
+        /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics/load)
+        #[wasm_bindgen(static_method_of = Atomics, catch)]
+        pub fn load(typed_array: &JsValue, index: u32) -> Result<f64, JsValue>;
+
+        /// The static `Atomics.notify()` method notifies up some agents that
+        /// are sleeping in the wait queue.
+        /// Note: This operation works with a shared `Int32Array` only.
```

Comment:
Since this only works with `Int32Array` I wonder if it would be useful to take `Int32Array` as the first argument actually? (same with `wait` and such I believe)